src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -39,7 +39,11 @@ public EditCommand parse(String args) throws ParseException {
         Index index;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            if (argMultimap.getPreamble().isEmpty()) {
+                index = null;
+            } else {
+                index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            }
         } catch (ParseException pe) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
```

Comment:
this looks a little nested, but there's no easy way to fix it. what I can think of is to put this in another function so that guard clauses can be used